### DISCUSSION TITLE
Site.js: another way of implementing TLS/HTTPS for your Owncast server

### DISCUSSION THREAD

**Discussion by aral** [owncast/owncast#926] [General Owncast discussion]
Hey folks,

So I was setting up my own Owncast instance today and realised that [Site.js](https://sitejs.org) would be perfect for the TLS proxy so I prepared a new release of it (version 16.2.0, released today) that allows you to create proxy servers in production (I hadn’t personally had a need for it until now).

## To implement TLS/HTTPS for your Owncast instance using Site.js:

1. Install Site.js: 
    ```shell
    wget -qO- https://sitejs.org/install | bash
    ```
2. `site enable :8080`
3. There is no three.

At Step 2, you will get a proxy server (systemd daemon) running at https://your.hostname and proxying HTTP and WS from your Owncast instance at port 8080. (The first time you hit the server, your Let’s Encrypt certificates will be automatically provisioned for you. They will also be automatically renewed.)

Enjoy! :)

**Comment by gabek** [comment]
This is awesome!

**Reply by aral** [reply]
Thanks, Gabe :) And wait for it, it’s about to get even better (but first, breakfast) ;)
